packages/flutter_tools/test/general.shard/devfs_test.dart

```diff
@@ -69,17 +69,22 @@ void main() {
       file.parent.createSync(recursive: true);
       file.writeAsBytesSync(<int>[1, 2, 3], flush: true);
 
-      final DateTime fiveSecondsAgo = DateTime.now().subtract(const Duration(seconds:5));
-      expect(content.isModifiedAfter(fiveSecondsAgo), isTrue);
-      expect(content.isModifiedAfter(fiveSecondsAgo), isTrue);
+      final DateTime fiveDaysAgo = DateTime.now().subtract(const Duration(days: 5));
+      expect(content.isModifiedAfter(fiveDaysAgo), isTrue);
+      expect(content.isModifiedAfter(fiveDaysAgo), isTrue);
       expect(content.isModifiedAfter(null), isTrue);
 
+      // Make sure some real time passes so the time stamp will be up to date.
+      // On a fast enough system with just the right timing, the two calls to
+      // file.writeasBytesSync will execute close-enough-together such that the
+      // timestamp will not actually get updated.
+      await Future<void>.delayed(const Duration(milliseconds: 500));
```

Comment:
Alternatively, I could see about making this test use FakeAsync.

Reply:
MemoryFileSystem.test ?

Reply:
I'm not sure if that will work though because MemoryFileSystem directly access DateTime

Reply:
It says: `The clock will increase monotonically each time it is used, disconnected from any real-world clock.`

https://api.flutter.dev/flutter/package-file_memory/MemoryFileSystem/MemoryFileSystem.test.html

We should probably switch all of the memfs usage to this

Reply:
Ok, figured out that I can use `MemoryFileSystem.test` which has the clock properties we really want.

Reply:
Hah, GitHub wasn't showing me your comments @jonahwilliams .  But yes, we should switch to this as much as possible.